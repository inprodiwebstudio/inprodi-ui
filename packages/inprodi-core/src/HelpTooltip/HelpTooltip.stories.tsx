import React                             from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { HelpTooltip, HelpTooltipProps } from "./HelpTooltip";

export default {
	title     : "Core/UI/HelpTooltip",
	component : HelpTooltip,
} as ComponentMeta<typeof HelpTooltip>;

const Template: ComponentStory<typeof HelpTooltip> = (args : HelpTooltipProps) => {

	return(
		<div style={{ width : "fit-content"}}>
			<HelpTooltip {...args} >
			Hola
			</HelpTooltip>
		</div>
	);
};

export const Basic = Template.bind({});
Basic.args = {
	label        : "Lable tooltip",
	position     : "left",
	withinPortal : false,
} as HelpTooltipProps;
