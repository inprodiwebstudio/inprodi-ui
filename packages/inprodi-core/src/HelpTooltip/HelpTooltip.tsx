import React           from "react";
import { HelpOutline } from "@inprodi/icons";
import {
	Box,
	BoxProps,
	Tooltip,
	TooltipProps,
	MantineNumberSize,
} from "@mantine/core";
import { FloatingPosition } from "@mantine/core/lib/Floating";

export interface HelpTooltipProps extends Omit<TooltipProps, "children"> {
	/**
	* Icon wrapper props
	*/
	boxProps ?: Omit<BoxProps, "children">;
	/**
	* HelpIcon size
	*/
    size ?: MantineNumberSize;
	/**
	* Tooltip position relative to target element (default) or mouse (floating)
	*/
	position ?: FloatingPosition;
}

export const HelpTooltip = ({ boxProps, size, ...rest } : HelpTooltipProps) => {
	return (
		<Tooltip
			multiline
			maw={280}
			position="left-end"
			{...rest}
		>
			<Box style={{ cursor : "pointer"}} {...boxProps}>
				<HelpOutline size={size}/>
			</Box>
		</Tooltip>
	);
};

export default HelpTooltip;
