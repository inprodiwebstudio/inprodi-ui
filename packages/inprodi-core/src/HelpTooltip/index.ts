export { HelpTooltip } from "./HelpTooltip";
export type { HelpTooltipProps } from "./HelpTooltip";
