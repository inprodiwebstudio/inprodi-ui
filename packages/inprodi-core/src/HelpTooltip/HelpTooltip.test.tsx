import React           from "react";
import { render }      from "@testing-library/react";
import { HelpTooltip } from "./HelpTooltip";

describe("HelpTooltip", () => {
	test("snaptshot", () => {
		const { container } = render(
			<HelpTooltip
				label="Help Label"
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
