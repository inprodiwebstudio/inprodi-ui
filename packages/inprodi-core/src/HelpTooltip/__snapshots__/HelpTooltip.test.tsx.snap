// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpTooltip snaptshot 1`] = `
<div>
  <div
    class="mantine-1avyp1d"
    style="cursor: pointer;"
  >
    <svg
      fill="none"
      height="16"
      viewBox="0 0 14 15"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 1.34a5.84 5.84 0 0 0-5.833 5.834A5.84 5.84 0 0 0 7 13.007a5.84 5.84 0 0 0 5.833-5.833A5.84 5.84 0 0 0 7 1.34Zm0 .875a4.952 4.952 0 0 1 4.958 4.959A4.952 4.952 0 0 1 7 12.132a4.952 4.952 0 0 1-4.958-4.958A4.952 4.952 0 0 1 7 2.215Zm0 1.75c-.961 0-1.75.789-1.75 1.75v.146a.438.438 0 1 0 .875 0v-.146c0-.488.387-.875.875-.875s.875.387.875.875c0 .683-.178.769-.489 1.013-.155.122-.36.263-.534.5-.174.238-.29.57-.29.966a.438.438 0 1 0 .875 0c0-.248.049-.35.12-.447.072-.098.197-.195.37-.33.344-.272.823-.78.823-1.702 0-.961-.789-1.75-1.75-1.75Zm0 5.542a.583.583 0 1 0 0 1.167.583.583 0 0 0 0-1.167Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
